cypress/e2e/10-undo-redo.cy.ts

```diff
@@ -38,17 +38,14 @@ describe('Undo/Redo', () => {
 	it('should undo/redo adding node in the middle', () => {
 		WorkflowPage.actions.addNodeToCanvas(SCHEDULE_TRIGGER_NODE_NAME);
 		WorkflowPage.actions.addNodeToCanvas(CODE_NODE_NAME);
-		WorkflowPage.actions.addNodeToCanvas(SET_NODE_NAME);
-		WorkflowPage.getters.nodeConnections().realHover();
-		cy.get('.connection-actions .add').filter(':visible').click();
-		WorkflowPage.actions.addNodeToCanvas(CODE_NODE_NAME, false);
+		WorkflowPage.actions.addNodeBetweenFirstTwoNodes(CODE_NODE_NAME);
```

Comment:
love the new functions.. might be more complicated but might be more reusable if you could specify where to inject

Reply:
That could be an interesting thing to implement. Hold my beer.

Reply:
Alright, this failed quickly :D
Decided not to do this right now since element order in HTML document is not reflecting their order on screen so getting corresponding connection for given node would add more complexity than we may want here.

Reply:
ya I guessed.. but at some point when we want to build more complex setups we gotta figure this out

Reply:
@mutdmour Done something in that direction here: https://github.com/n8n-io/n8n/pull/5393